### DISCUSSION TITLE
Mock ClientSet

### DISCUSSION THREAD

**Discussion by Hunter-Thompson** [argoproj/argo-workflows#7568] [Q&A] [answered]
hey everyone, Im trying to test a Go application which creates workflows, it uses the Go SDK.

Is there a way to get a mock client set that I can use to dry run creating workflows? 

For e.g, with Kubernetes client-go I can do `fake.NewSimpleClientSet()`

The function that creates the workflow accepts `workflowpkg.WorkflowServiceClient` as an argument.

func:

```go
func CreateWorkflow(wfRequest *workflow.WorkflowCreateRequest, WFClient workflowpkg.WorkflowServiceClient) (string, error) {

	wf, err := WFClient.CreateWorkflow(context.Background, wfRequest)
	if err != nil {
		return "", err
	}

	return wf.GetName(), nil
}
```

**Comment by terrytangyuan** [comment, marked as answer]
This might be what you are looking for: https://github.com/argoproj/argo-workflows/blob/303bf79f94c5b09e4f25add23c44ed2ab1536039/pkg/client/clientset/versioned/fake/clientset_generated.go#L20

**Reply by Hunter-Thompson** [reply]
Thanks, this was it.

Also genius to have a fake client : )
